host/usb/src/hub.c

```diff
@@ -391,8 +418,10 @@ static void ext_port_event_callback(ext_port_hdl_t port_hdl, ext_port_event_t ev
 #endif // ENABLE_USB_HUBS
 
 // ---------------------- Handlers -------------------------
-static void root_port_handle_events(hcd_port_handle_t root_port_hdl)
+static void root_port_handle_events(root_port_t *root_port)
 {
+    hcd_port_handle_t root_port_hdl = root_port->constant.hdl;
+    uint8_t port_idx = hcd_port_get_number(root_port_hdl);
```

Comment:
You could use `const` to indicate the port_index is not changing in the function

```suggestion
    const uint8_t port_idx = hcd_port_get_number(root_port_hdl);
```